roles/docker/vars/ubuntu.yml

```diff
@@ -1,15 +1,13 @@
 ---
 docker_kernel_min_version: '3.10'
+docker_package_arch: 'amd64'
 
-# https://apt.dockerproject.org/repo/dists/ubuntu-xenial/main/filelist
+# https://download.docker.com/linux/ubuntu/dists/trusty/stable/
 docker_versioned_pkg:
-  'latest': docker-engine
-  '1.11': docker-engine=1.11.1-0~{{ ansible_distribution_release|lower }}
-  '1.12': docker-engine=1.12.6-0~ubuntu-{{ ansible_distribution_release|lower }}
-  '1.13': docker-engine=1.13.1-0~ubuntu-{{ ansible_distribution_release|lower }}
-  '17.03': docker-engine=17.03.1~ce-0~ubuntu-{{ ansible_distribution_release|lower }}
-  'stable': docker-engine=17.03.1~ce-0~ubuntu-{{ ansible_distribution_release|lower }}
-  'edge': docker-engine=17.05.0~ce-0~ubuntu-{{ ansible_distribution_release|lower }}
+  'latest': docker-ce
+  '17.12': docker-ce=17.03.1~ce-0~ubuntu
+  'stable': docker-ce=17.12.0~ce-0~ubuntu
```

Comment:
Why choose this version of docker？
You can see kubernetes1.8 and 1.9 for the docker version of the instructions
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md#v193
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.8.md
The release changelog for kubernetes 1.9 gives a stable version of docker.
(The validated docker versions are the same as for v1.8: 1.11.2 to 1.13.1 and 17.03.x)
So I think we should choose 17.03.x version, such as 17.03.2 version, this version is fully verified in kubernetes1.8.

Reply:
Agree, let's follow the recommended version from kubernetes. so 17.03

Reply:
Definitely my bad on the typo, however unfortunately it doesn't appear the new repo goes back very far at all - see also https://download.docker.com/linux/ubuntu/dists/artful/pool/stable/amd64/

Given that this new apt repo is entirely focused on 17.11+, we may want to close this PR until k8s moves to the later releases...

Reply:
It's link for Artful. But kubespray supports only Xenial, as I know.
New apt repo has all needed packages - https://download.docker.com/linux/ubuntu/dists/xenial/pool/stable/amd64/